elastic/kibana
# [Discover] Extend DiscoverNoResults component to show different message on error

## Summary

With the alignment of error notification in PR #77788 all error messages in Discover were migrated to be a toast. This could be misleading when there's an error and the no-results page is displayed:

<img width="1257" alt="Bildschirmfoto 2020-10-06 um 16 06 04" src="https://user-images.githubusercontent.com/463851/95212947-8d599000-07ee-11eb-9860-8bb965eb7dc9.png">

This PR displays a different message in the error case

![Bildschirmfoto 2020-10-20 um 09 57 17](https://user-images.githubusercontent.com/463851/96557637-d1529780-12ba-11eb-96b8-d1cd562e8330.png)


Furthermore this PR converts `DiscoverNoResults` to TypeScript, modularizes the component, migrates testing to no longer use snapshots of the whole component, and moves the whole component out of the `angular` folder.

## Testing

1. Add an index pattern containing a invalid painless script by importing the saved object in management [export.ndjson.zip](https://github.com/elastic/kibana/files/5384971/export-19.ndjson.zip)
2. This adds the `logs*` index pattern. Add data to the `logs` index 
```
PUT log/_doc/1
{
  "@timestamp":"2020-10-14T07:07:29.952Z",
  "message": "test"
}
```
3. Navigate to Discover, select the `logs` index pattern, adapt time range
4. Submit and then you should see the error message 

### Checklist
